src/sortStudents.ts

```diff
@@ -1,16 +1,60 @@
 
 export interface Student {
   // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
+
 }
 
 export enum SortType {
   // describe SortType enum
+  Name = 'n',
+  Surname = 's',
+  Age = 'a',
+  Married = 'm',
+  AverageGrade = 'aG',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(students: Student[],
+  sortBy: SortType, order: SortOrder): Student[] {
+  const sortedStudents = [...students];
+
+  sortedStudents.sort((a, b) => {
+    let result = 0;
+
+    switch (sortBy) {
+      case SortType.Name:
+        result = a.name.localeCompare(b.name);
+        break;
+      case SortType.Surname:
+        result = a.surname.localeCompare(b.surname);
+        break;
```

Comment:
you can combine similar cases since the logic is pretty much the same
BUT for that you would have to refactor your enum so it allows you accessing the object properties - see the comment above
```suggestion
      case SortType.Name:
      case SortType.Surname:
        result = a[sortBy].localeCompare(b[sortBy]);
        break;
```